### ISSUE TITLE
question: flag unused imports?

### ISSUE THREAD

**Issue by tony-kerz** [eslint/eslint#4843]
i'm looking for a config to flag unused imports.

e.g. if i have a file like:

```
import _ from 'lodash'

export default function doesNotUseLodash() {
  console.log('does nothing)
}
```

i would want to see the fact that i'm importing `_` and not using it flagged.

here is my current [`.eslintrc.js` file](https://github.com/tony-kerz/redux-scratch/blob/d987433f1da4929d86a9764c7042f0c0c67be953/.eslintrc.js).

i'm new to eslint, so easily possible i'm just being a dunce.

any guidance appreciated!

best,
tony.

**Comment by eslintbot** [comment]
@tony-kerz Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by gyandeeps** [comment]
`no-unused-vars` rule should catch this. I tried it using eslint online [demo](http://eslint.org/demo/) and it works fine.
Now I do see that you are using `babel-eslint` parser so I am not sure if thats creating an issue or not.

**Comment by nzakas** [comment]
Yeah, `no-unused-vars: 2` is already set in your config, so that should pick up unused imports. 

If you're not seeing an error, then try using the default parser instead of `babel-eslint`.

**Comment by tony-kerz** [comment]
thanks guys.

fyi, it turns out the issue was that eslint wasn't looking at `.jsx` files cuz of operator error.

i was able to get expected behavior with the following two fixes:

1. in `package.json`, convert:

  ```
  "scripts": {
    "lint": "eslint app"
  },
  ```
to:  
  ```
  "scripts": {
    "lint": "eslint --ext .js --ext .jsx app"
  },
  ```
1. in `webpack.config.js`, convert:

  ```
    preLoaders: [
      {test: /\.js$/, loader: 'eslint-loader', exclude: /node_modules/}
    ],
  ```
to:  
  ```
    preLoaders: [
      {test: /\.jsx?$/, loader: 'eslint-loader', exclude: /node_modules/}
    ],
  ```
regards,
tony.
